### PR TITLE
Use bit-pattern comparison when meeting float approximations (flambda)

### PR DESCRIPTION
**Pull request by xclerc**
Fixes the bug reported by @Chris00 in https://github.com/ocaml/ocaml/pull/1794.

The bug is that flambda used `Stdlib.(=)` to compare float values
(thus considering `-0.` and `+0.` equal) while applying a transform
that simplifies `if ... then e1 else e2` when `e1` and `e2` have the
same approximation.

(If time permits, I think it should be considered for inclusion in 4.07.)

### CODE REVIEW

**Review by nojb — APPROVED** [review]
Patch seems clearly correct. LGTM.

File: middle_end/simple_value_approx.ml [hunk 1]
```diff
@@ -655,6 +655,12 @@ let equal_boxed_int (type t1) (type t2)
   | Nativeint, Nativeint -> Nativeint.equal i1 i2
   | _ -> false
 
+let compare_floats f1 f2 =
```
Comment (gasche) [inline middle_end/simple_value_approx.ml]: This should be named `equal_floats`.

**Comment by nojb** [comment]
Personally I think this is safe for inclusion in 4.07. @damiendoligez?

File: middle_end/simple_value_approx.ml (hunk 1 above)
Comment (xclerc) [reply middle_end/simple_value_approx.ml]: Indeed, thanks.

**Comment by gasche** [comment]
Merged and cherry-picked in 4.07.

**Pull request merged**